### PR TITLE
Changed spinlock behavior for ipi to fix deadlock

### PR DESCRIPTION
**Pull request by ThaumicMekanism**
This is one of the initial fixes to the deadlock issues for multi-core usage.

### CODE REVIEW

File: sm/pmp.c [hunk 1]
```diff
@@ -252,7 +266,7 @@ int pmp_unset_global(int region_idx)
   /* We avoid any complex PMP-related IPI management
    * by ensuring only one hart can enter this region at a time */
 #ifdef __riscv_atomic
-  spinlock_lock(&pmp_ipi_global_lock);
+  pmp_ipi_acquire_lock();
   send_and_sync_pmp_ipi(region_idx, IPI_PMP_UNSET, PMP_NO_PERM);
   spinlock_unlock(&pmp_ipi_global_lock);
```
Comment (dayeol) [inline sm/pmp.c]: Could you wrap this with pmp_ipi_release_lock() for symmetricity?

File: sm/pmp.c [hunk 2]
```diff
@@ -238,6 +233,25 @@ static void send_and_sync_pmp_ipi(int region_idx, enum ipi_type type, uint8_t pe
...
+    } else {
+      pmp_unset(ipi_region_idx);
+    }
+
+    ipi_mailbox[read_csr(mhartid)].pending = 0;
+  }
+}
+
+void pmp_ipi_acquire_lock() {
+  while(spinlock_trylock(&pmp_ipi_global_lock)) {
```
Comment (dayeol) [inline sm/pmp.c:259]: Please add comment

**Review by dayeol — CHANGES_REQUESTED** [review]
Looks good, requesting small changes

File: sm/pmp.c (hunk 1 above)
Comment (ThaumicMekanism) [reply sm/pmp.c]: Ah ya, I was thinking about doing that as well. I had forgot to add it in. I will do that rn.

File: sm/pmp.c (hunk 2 above)
Comment (ThaumicMekanism) [reply sm/pmp.c:259]: I have just added comments to this function and the other functions which I made.

File: sm/pmp.c (hunk 1 above)
Comment (ThaumicMekanism) [reply sm/pmp.c]: Done!

**Review by dayeol — APPROVED** [review]
LGTM

**Pull request merged**